pkg/apps/alert/alert.go

```diff
@@ -94,6 +95,23 @@ func (a *Alert) getCreater(version string) (Creater, error) {
 	return nil, fmt.Errorf("version %s is not supported", version)
 }
 
+func (a *Alert) ensureVersion(alt *alertapi.Alert) error {
+	// If the version is not specified then we set it to be the latest.
+	versions := a.Versions()
+	if len(alt.Spec.Version) == 0 {
+		sort.Sort(sort.Reverse(sort.StringSlice(versions)))
+		alt.Spec.Version = versions[0]
+		return nil
+	}
+	// Check the provided version
```

Comment:
But this in an else block.  No point in checking for a version supplied when we set it ourselves.